CMakeLists.txt

```diff
@@ -1,3 +1,62 @@
+#
+# CMakeLists.txt - CMake configuration file for swmm/tests
+#
+# Created: February 23, 2018
+# Author:  Constantin Savtchenko
+# Ref: http://neyasystems.com/an-engineers-guide-to-unit-testing-cmake-and-boost-unit-tests/
+#
+# Modified by: Michael E. Tryby
+#              US EPA ORD/NRMRL
+# 
+# Modified by: Velitchko G. Tzatchkov
+#              IMTA (Mexican Institute of Water Technology)
+#	       June 29, 2020
+# 
+
+cmake_minimum_required (VERSION 3.0)
+
+
+#Setup CMake to run tests
+enable_testing()
+
+
+# Sets for output directory for executables and libraries. 
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+
+if(UNIX)
+    set(CMAKE_CXX_FLAGS "-std=c++11")
+endif(UNIX)
+
+
+#Prep ourselves for compiling boost
+#find_package(Boost REQUIRED)
+include_directories (${Boost_INCLUDE_DIRS})
+
+
+# Test files are a separate source directory called tests
+file(GLOB TEST_SRCS . tests/test_*.cpp)
+
+#add_library(coupling . src/coupling.c)
+
+#Run through each source
+foreach(testSrc ${TEST_SRCS})
+        #Extract the filename without an extension (NAME_WE)
+        get_filename_component(testName ${testSrc} NAME_WE)
+
+        #Add compile target
+        add_executable(${testName} ${testSrc})
+
+        #link to Boost libraries AND your targets and dependencies
+        #target_link_libraries(${testName} ${Boost_LIBRARIES} swmm5 swmm-output coupling)
+        target_link_libraries(${testName} ${Boost_LIBRARIES} swmm5 swmm-output)
+
+        #Finally add it to test execution - 
+        #Notice the WORKING_DIRECTORY and COMMAND
+        add_test(NAME ${testName}
+                 COMMAND ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${testName}  
+                 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/data)
+endforeach(testSrc)
```

Comment:
What is the reason of adding this large chunk of code? How is this related to the tests?

Reply:
@lrntct 
That chunk of code is actually the content of the CMakeLists.txt file that removed from the tests directory, in order to work with only one CMakeLists.txt (that in the root directory). It works either way, with one CMakeLists.txt or with several CMakeLists.txt.

Derived from your comment, however, I see that working with two separate CMakeLists.txt files is preferred in the repository, so that, in subsequent commits I removed that chunk of code and re-established the CMakeLists.txt in the tests directory.